gpflow/conditionals.py

```diff
@@ -111,13 +111,15 @@ def _conditional(Xnew, X, kern, f, *, full_cov=False, q_sqrt=None, white=False):
         - variance: N x R (full_cov = False), R x N x N (full_cov = True)
     """
     logger.debug("Conditional: Kernel")
-    num_data = tf.shape(X)[0]  # M
-    Kmm = kern.K(X) + tf.eye(num_data, dtype=settings.float_type) * settings.numerics.jitter_level
-    Kmn = kern.K(X, Xnew)
+    num_data = tf.shape(X)[-2]  # M
+    Kmm = kern.K(X) + tf.eye(num_data, dtype=settings.float_type) * settings.numerics.jitter_level  #  [..., M, M]
+    Kmn = kern.K(X, Xnew)  # [M, ..., N]
+
     if full_cov:
-        Knn = kern.K(Xnew)
+        Knn = kern.K(Xnew)  # [...,N,N]
```

Comment:
[discuss] These if else are ugly, just a note for next PRs: we need `kern.K(X, diag=not full_cov)` or `kern.K(X, full=full_cov)`

Reply:
yes, I agree. We should add this to our todo list